pkg/osquery/runtime/runtime.go

```diff
@@ -524,6 +525,32 @@ func (r *Runner) launchOsqueryInstance() error {
 		return errors.Wrap(err, "could not calculate osquery file paths")
 	}
 
+	// TODO: should also handle windows. probably
+	if runtime.GOOS != "windows" {
+		// The extensions file should be owned by the process's UID or the file
+		// should be owned by root. Osquery will refuse to load the extension
+		// otherwise
+		fd, err := os.Stat(paths.extensionPath)
+		if err != nil {
+			return errors.Wrap(err, "stat-ing extension path")
+		}
+		sys := fd.Sys().(*syscall.Stat_t)
+		isRootOwned := (sys.Uid == 0)
+		isProcOwned := (sys.Uid == uint32(os.Geteuid()))
+
+		if !(isRootOwned || isProcOwned) {
+			level.Info(o.logger).Log(
+				"event", "BCJ_",
+				"msg", "unsafe permissions detected on extension binary")
+
+			// chown the file
+			err := os.Chown(paths.extensionPath, os.Getuid(), os.Getgid())
```

Comment:
We'll meed to test, but my inclination is that we probably need to chown through all of the root dir